src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py

```diff
@@ -0,0 +1,261 @@
+
+
+def randn_heur_block(args):
+    if args["N"] <= 512:
+        return 512
+    else:
+        return 1024
+
+
+def randn_heur_num_warps(args):
+    if args["N"] <= 512:
+        return 4
+    elif args["N"] <= 1024:
+        return 8
+    else:
+        return 16
+
+
+def softmax_heur_tile_k(args):
+    MAX_TILE_K = 8192
+    NUM_SMS = torch.cuda.get_device_properties(
+        torch.cuda.current_device()
+    ).multi_processor_count
+    tile_k = 1
+    upper_bound = min(args["K"], MAX_TILE_K)
+    while tile_k <= upper_bound:
+        num_blocks = args["M"] * triton.cdiv(args["K"], tile_k)
+        num_waves = num_blocks / NUM_SMS
+        if (num_waves > 1) and (tile_k * 2 <= upper_bound):
+            tile_k *= 2
+        else:
+            break
+    return tile_k
+
+
+def softmax_heur_tile_n_non_inner(args):
+    return triton.cdiv(8192, args["TILE_K"])
+
+
+def softmax_heur_one_tile_per_cta(args):
+    return args["TILE_N"] >= args["N"]
+
+
+def softmax_heur_num_warps_non_inner(args):
+    tile_size = args["TILE_N"] * args["TILE_K"]
+    if tile_size < 2048:
+        return 4
+    elif tile_size < 4096:
+        return 8
+    else:
+        return 16
+
+
+def softmax_heur_tile_n_inner(args):
+    if args["N"] <= (32 * 1024):
+        return triton.next_power_of_2(args["N"])
+    else:
+        return 4096
+
+
+def softmax_heur_num_warps_inner(args):
+    tile_size = args["TILE_N"]
+    if tile_size < 2048:
+        return 4
+    elif tile_size < 4096:
+        return 8
+    else:
+        return 16
+
+
+def softmax_heur_tile_n_bwd_non_inner(args):
+    return max(1, 1024 // args["TILE_K"])
+
+
+def softmax_heru_tile_m(args):
+    return max(1, 1024 // args["TILE_N"])
+
+
+def uniform_heur_block(args):
+    if args["N"] <= 512:
+        return 512
+    else:
+        return 1024
+
+
+def uniform_heur_num_warps(args):
+    if args["N"] <= 512:
+        return 4
+    elif args["N"] <= 1024:
+        return 8
+    else:
+        return 16
+
+
+def var_mean_heur_block_n(args):
+    return triton.next_power_of_2(args["BLOCK_NUM"])
+
+
+HEURISTICS_CONFIGS = {
+    "argmax": {
+        "BLOCK_M": argmax_heur_block_m,
+        "BLOCK_N": argmax_heur_block_n,
+    },
+    "bmm": {
+        "DIVISIBLE_M": bmm_heur_divisible_m,
+        "DIVISIBLE_N": bmm_heur_divisible_n,
+        "DIVISIBLE_K": bmm_heur_divisible_k,
+    },
+    "dropout": {
+        "BLOCK": dropout_heur_block,
+        "num_warps": dropout_heur_num_warps,
+    },
+    "exponential_": {
+        "BLOCK": exponential_heur_block,
+        "num_warps": exponential_heur_num_warps,
+    },
+    "gather": {
+        "BLOCK_M": gather_heur_block_m,
+        "BLOCK_N": gather_heur_block_n,
+    },
+    "index_select": {
+        "BLOCK_M": index_select_heur_block_m,
+        "BLOCK_N": index_select_heur_block_n,
+    },
+    "mm": {
+        "EVEN_K": mm_heur_even_k,
+    },
+    "multinomial": {
+        "NBLOCK": lambda args: 128,
+        "num_warps": lambda args: 4,
```

Comment:
set the constant value as default

Reply:
Done

Reply:
defining these lambda expressions as functions.